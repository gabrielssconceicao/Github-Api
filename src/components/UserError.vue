<script setup lang="ts"></script>
<template>
  <section
    class="flex flex-col items-center justify-center flex-1 gap-3 max-w-[400px] min-h-60 bg-white rounded-lg font-mono px-1">
    <h2 class="text-3xl font-bold  text-red-600 mb-4">User Not Found
    </h2>
    <p class="text-gray-800 font-semibold text-center">Sorry, we couldn't find the user you were looking for.</p>
  </section>
</template>
<style>
h2 {
  text-shadow: 2px 1px 2px #00000088;
}
</style>
