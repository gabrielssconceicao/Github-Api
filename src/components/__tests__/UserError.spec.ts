import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import UserError from '../UserError.vue';

describe('<UserError />', () => {
  it('should render', () => {
    const wrapper = mount(UserError);
    expect(wrapper.findComponent(UserError).exists()).toBe(true);

    expect(wrapper.find('h2').exists()).toBe(true);
    expect(wrapper.find('h2').text()).toBe('User Not Found');

    expect(wrapper.find('p').exists()).toBe(true);
    expect(wrapper.find('p').text()).toBe(
      "Sorry, we couldn't find the user you were looking for.",
    );
  });

  it('should match snapshot', () => {
    const wrapper = mount(UserError);
    expect(wrapper.element).toMatchSnapshot();
  });
});
