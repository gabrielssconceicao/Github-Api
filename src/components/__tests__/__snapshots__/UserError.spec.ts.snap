// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserError /> > should match snapshot 1`] = `
<section
  class="flex flex-col items-center justify-center flex-1 gap-3 max-w-[400px] h-[60%] bg-white rounded-lg font-mono px-1"
>
  <h2
    class="text-3xl font-bold text-red-600 mb-4"
  >
    User Not Found 
  </h2>
  <p
    class="text-gray-800 font-semibold text-center"
  >
    Sorry, we couldn't find the user you were looking for.
  </p>
</section>
`;
